ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/classload/ImportsManager.java

```diff
@@ -303,5 +317,12 @@ private void storeAnonImplicitPrefixes() {
     private void storeImport(QuotedIdentifier quotedPrefix, QuotedImport moduleName) {
         this.imports.put(quotedPrefix, moduleName);
         this.reverseImports.put(moduleName, quotedPrefix);
+        if (!quotedPrefix.toString().equals(JAVA_PREFIX)) {
+            this.userImports.put(quotedPrefix, moduleName);
+        } else {
+            if (this.initialized.get()) {
+                this.userImports.put(quotedPrefix, moduleName);
+            }
+        }
```

Comment:
Can't this be simplified to the following? 🤔 
```java
if (!quotedPrefix.toString().equals(JAVA_PREFIX) || this.initialized.get()) {
     this.userImports.put(quotedPrefix, moduleName);
}
```

Reply:
Will go ahead and merge for now. If the above is possible, let's address in a separate PR.